client/transaction-pool/api/src/lib.rs

```diff
@@ -289,8 +289,10 @@ pub enum ChainEvent<B: BlockT> {
 	},
 	/// An existing block has been finalized.
 	Finalized {
-		/// Hash of just finalized block
+		/// Hash of just finalized block.
 		hash: B::Hash,
+		/// Path from old finalized to new finalized parent.
+		tree_route: Arc<Vec<B::Hash>>,
```

Comment:
We should merge both imo. There is no real need to just have the latest finalized block.